pennylane/devices/qubit/simulate.py

```diff
@@ -124,24 +134,34 @@ def get_final_state(circuit, debugger=None, interface=None):
 
     # initial state is batched only if the state preparation (if it exists) is batched
     is_state_batched = bool(prep and prep.batch_size is not None)
+    measurement_values = {}
     for op in circuit.operations[bool(prep) :]:
-        state = apply_operation(op, state, is_state_batched=is_state_batched, debugger=debugger)
-
+        if isinstance(op, Conditional):
```

Comment:
I know my earlier comment about the return of `apply_operation` forced us to stop handling `MidMeasureMP` with `apply_operation`.

Feel free to ignore this comment, as it is optional.

But we could instead add an optional keyword argument to `apply_operation`, and then have `apply_operation` handle both `Conditional` and `MidMeasureMP`.  They would both just rely on the mutable keyword argument `measurement_values`.


```
state = apply_operation(op, state, is_state_batched=is_state_batched, debugger=debugger, mid_measurements=mid_measurements)
```

Basically, we don't have to add an additional output if we are mutating an input.